ui/app/serializers/sync/association.js

```diff
@@ -31,4 +32,32 @@ export default class SyncAssociationSerializer extends ApplicationSerializer {
     }
     return payload;
   }
+
+  normalizeFetchByDestinations(payload) {
+    const { store_name, store_type, associated_secrets } = payload.data;
+    const unsynced = [];
+    let lastSync;
+
+    for (const key in associated_secrets) {
+      const association = associated_secrets[key];
+      // for display purposes, any status other than SYNCED is considered unsynced
+      if (association.sync_status !== 'SYNCED') {
+        unsynced.push(association.sync_status);
+      }
+      // use the most recent updated_at value as the last synced date
+      const updated = new Date(association.updated_at);
+      if (!lastSync || updated > lastSync) {
+        lastSync = updated;
+      }
+    }
+
+    return {
+      icon: findDestination(store_type).icon,
+      name: store_name,
+      type: store_type,
+      associationCount: Object.entries(associated_secrets).length,
+      status: unsynced.length ? `${unsynced.length} Unsynced` : 'All synced',
```

Comment:
We also need to account for if there are no secrets at all

Reply:
Good call! I think hiding the badge is probably appropriate for that state.

Reply:
Added a destination with no associations to the mirage scenario after making the changes so that it's represented. 

![image](https://github.com/hashicorp/vault/assets/24611656/d5bdbb3a-72a1-41a6-a0f7-bb438fa601cc)